platform/drm/cog-platform-drm.c

```diff
@@ -494,19 +509,30 @@ init_drm(void)
             (long)((drm_data.mode - drm_data.connector.obj->modes) / sizeof(drmModeModeInfo *)), drm_data.mode->name,
             drm_data.mode->vrefresh);
 
+    /* Try the currently connected encoder+crtc */
     for (int i = 0; i < drm_data.base_resources->count_encoders; ++i) {
         drm_data.encoder = drmModeGetEncoder(drm_data.fd, drm_data.base_resources->encoders[i]);
-        if (drm_data.encoder->encoder_id == drm_data.connector.obj->encoder_id)
+        if (!drm_data.encoder) {
+            /* cannot retrieve encoder, ignoring... */
+            continue;
+        }
+
+        const int32_t crtc_id = find_crtc_for_encoder(drm_data.base_resources, drm_data.encoder);
+        if (crtc_id != 0) {
```

Comment:
Either the `find_crtc_for_encoder()` is changed to return zero above, or this should check `crtc_id != -1`.

Reply:
fixed with != -1